android/app/src/main/java/com/allaboutolaf/BalancesIntentService.java

```diff
@@ -0,0 +1,60 @@
+package com.allaboutolaf;
+
+import android.app.IntentService;
+import android.content.Intent;
+import android.os.StrictMode;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import com.squareup.okhttp.MediaType;
+import com.squareup.okhttp.OkHttpClient;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.RequestBody;
+import com.squareup.okhttp.Response;
+
+import java.io.IOException;
+import java.util.logging.Logger;
+
+/**
+ * Created by everdoorn on 8/8/17.
+ */
+
+public class BalancesIntentService extends IntentService {
```

Comment:
I've been thinking about this, and really doing stuff from the JS code and passing to the native via a file is not a good solution. We want the widget to update without the user having to start the app, so we have to go native. This is the start of a native service implementation that should allow the widget to update in the background. I'm a tad stuck with the authentication, though. Do either @hawkrives or @drewvolz know how we might pass the user's info that we store in the keychain to the native system? I haven't poked around the source of that package yet.